libraries/Microsoft.Bot.Builder.Core/Adapters/TestAdapter.cs

```diff
@@ -434,6 +443,47 @@ public TestFlow Test(string userSays, Action<IActivity> expected, string descrip
                 .AssertReply(expected, description, timeout);
         }
 
+        /// <summary>
+        /// Shorcut to test a conversation for .Send(user).AssertReply(bot)
+        /// Each activity with From.Role equals to "bot" will be processed with AssertReply mthod
+        /// Every other activity will be processed as User's message with Send method
+        /// </summary>
+        /// <param name="activities">List of activities to test</param>
+        /// <param name="description"></param>
+        /// <param name="timeout"></param>
+        /// <returns>TestFlow to test</returns>
+        public TestFlow Test(IEnumerable<IActivity> activities, [CallerMemberName] string description = null, UInt32 timeout = 3000)
+        {
+            if (activities == null)
+                throw new ArgumentNullException(nameof(activities));
+
+            bool IsReply(IActivity activity) => string.Equals("bot", activity.From?.Role, StringComparison.InvariantCultureIgnoreCase);
+
+            return activities.Aggregate(this, (flow, activity) => IsReply(activity) ? flow.AssertReply(activity, description, timeout) : flow.Send(activity));
```

Comment:
Could you comment & simplify this chunk of code? 

I'm sure it's obvious now what it's doing, but in 6 months it won't be....